mitiq/cdr/cdr_tests.py

```diff
@@ -1,121 +1,134 @@
-from clifford_training_data import circuit_to_array, array_to_circuit, generate_training_circuits, is_clifford_angle
+from qiskit import QuantumCircuit, compiler
+from clifford_training_data import (count_non_cliffords,
+                                    generate_training_circuits)
 import numpy as np
+import matplotlib.pyplot as plt
+
 
 def random_circuit(
-    qubits: int, 
+    qubits: int,
     depth: int,
-    measure: bool,
-)-> QPROGRAM:
-
-    '''Function to generate a random quantum circuit in cirq. The circuit is based on the hardware efficient ansatz,
-    with alternating CNOT layers with randomly selected single qubit gates in between.
-    
-    Args: 
-        qubits: number of qubits in circuit. 
+) -> QPROGRAM:
+    '''Function to generate a random quantum circuit in cirq. The circuit is \
+       based on the hardware efficient ansatz,
+    with alternating CNOT layers with randomly selected single qubit gates in \
+    between.
+    Args:
+        qubits: number of qubits in circuit.
         depth: depth of the RQC.
-        measure: measurements or not.
-        
-    Returns: 
+    Returns:
         cirquit: a random quantum circuit of specified depth.
-        
     '''
     # Get a rectangular grid of qubits.
     qubits = cirq.GridQubit.rect(qubits, 1)
     # Generates a random circuit on the provided qubits.
-    circuit = cirq.experiments.random_rotations_between_grid_interaction_layers_circuit(
-        qubits = qubits, depth=depth, seed=0)
-    if measure == True:
-        circuit.append(cirq.measure(*qubits, key='z'))
+    circuit = \
+        cirq.experiments.\
+        random_rotations_between_grid_interaction_layers_circuit(
+            qubits=qubits, depth=depth, seed=0)
+    circuit.append(cirq.measure(*qubits, key='z'))
     return(circuit)
 
+
 def qiskit_circuit_transplation(
     circ: QPROGRAM,
-)-> QPROGRAM:
-    """Decomposes qiskit circuit object into Rz, Rx(pi/2) (sx), X and CNOT gates. 
-    
+) -> QPROGRAM:
+    """Decomposes qiskit circuit object into Rz, Rx(pi/2) (sx), X and CNOT \
+       gates.
     Args:
         circ: original circuit of interest assumed to be qiskit circuit object.
-        
-    Returns: 
+    Returns:
         circ_new: new circuite compiled and decomposed into the above gate set.
     """
-    # this decomposes the circuit into u3 and cnot gates:
-    circ = compiler.transpile(circ, basis_gates = ['sx','rz','cx', 'x'], optimization_level=3)
-    #print(circ.draw())
-    # now for each U3(theta, phi, lambda), this can be converted into Rz(phi+pi)Rx(pi/2)Rz(theta+pi)Rx(pi/2)Rz(lambda)
-    try:
-        circ_new = QuantumCircuit(len(circ.qubits), len(circ.clbits))
-    except:
-        circ_new = QuantumCircuit(len(circ.qubits))
+    circ = compiler.transpile(circ, basis_gates=['u3', 'cx'],
+                              optimization_level=3)
+    circ_new = QuantumCircuit(len(circ.qubits), len(circ.clbits))
     for i in range(len(circ.data)):
```

Comment:
`for (gate, qubits, cbits) in circ.data` is a cleaner way to iterate here I think. There's a lot of repeated code in the for loop getting gate name, qubits, etc.

Reply:
Yes that is much better. Will change.